automated/linux/pkcs11/pkcs11.sh

```diff
@@ -0,0 +1,341 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (C) 2021 Foundries.io Ltd.
+
+# shellcheck disable=SC1091
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+export RESULT_FILE
+PTOOL="pkcs11-tool --module /usr/lib/libckteec.so.0.1"
+USE_SE05X="True"
+EXECUTE_LOOP="True"
+
+usage() {
+    echo "\
+    Usage: $0 [-p <pkcs11-tool>] [-t <true|false>] [-s <true|false>] [-l <true|false>]
+
+    -p <pkcs11-tool>
+        pkcs11-tool with all the options required. Default is:
+        pkcs11-tool --module /usr/lib/libckteec.so.0.1
+    -t <true|false>
+        This flag switches on/off the use of SE05X TPM device.
+        It is turned on by default but users can choose to use
+        softhsm as alternative. Setting this flag to False will
+        prevent script from calling ssscli.
+    -l <true|false>
+        Run loop test. The test is meant to only be executed on
+        SE05x device. Running this test with sofhsm might take
+        a very long time and exhaust memory/storage on the device.
+    -s <true|false>
+        Skip install. True by default.
+    "
+}
+
+while getopts "p:t:s:l:h" opts; do
+    case "$opts" in
+        p) PTOOL="${OPTARG}";;
+        t) USE_SE05X="${OPTARG}";;
+        s) SKIP_INSTALL="${OPTARG}";;
+        l) EXECUTE_LOOP="${OPTARG}";;
+        h|*) usage ; exit 1 ;;
+    esac
+done
+
+! check_root && error_msg "You need to be root to run this script."
+# Test run.
+create_out_dir "${OUTPUT}"
+install_deps "opensc bc" "${SKIP_INSTALL}"
```

Comment:
should 'openssl' be part of the install_deps?

Reply:
I think you're right, I'll add it. The ssscli doesn't have a package in any distro (as far as I'm aware).